tests/integration/acp/demo_test.go

```diff
@@ -0,0 +1,137 @@
+
+		Description: "DEMO",
+
+		Actions: []any{
+			testUtils.AddPolicy{ // Specified and subbed in for User1 and User2 schema below
+
+				Identity: immutable.Some(1),
+
+				Policy: `
+                    name: test
+                    description: a test policy which marks a collection in a database as a resource
+
+                    actor:
+                      name: actor
+
+                    resources:
+                      users:
+                        permissions:
+                          read:
+                            expr: owner + reader
+                          write:
+                            expr: owner
+
+                        relations:
+                          owner:
+                            types:
+                              - actor
+                          reader:
+                            types:
+                              - actor
+                          admin:
+                            manages:
+                              - reader
+                            types:
+                              - actor
+                `,
+
+				// TODO: Remove after draft is approved
+				// Note: This is also valid now that it is optional, but we don't need to provide it at all.
+				ExpectedPolicyID: immutable.Some(
+					"94eb195c0e459aa79e02a1986c7e731c5015721c18a373f2b2a0ed140a04b454",
+				),
+			},
+
+			testUtils.AddPolicy{ // Specified and subbed in for User3 schema below
+
+				Identity: immutable.Some(1),
+
+				Policy: `
+                    name: test
+                    description: another policy
+
+                    actor:
+                      name: actor
+
+                    resources:
+                      users:
+                        permissions:
+                          read:
+                            expr: owner + reader
+                          write:
+                            expr: owner
+
+                        relations:
+                          owner:
+                            types:
+                              - actor
+                          reader:
+                            types:
+                              - actor
+                          admin:
+                            manages:
+                              - reader
+                            types:
+                              - actor
+                `,
+
+				// TODO: Remove after draft is approved
+				// Note: No policyID assertion needed
+			},
+
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users1 @policy(
+						id: "%policyID%",
+						resource: "users"
+					) {
+						name: String
+						age: Int
+					}
+
+					type User2 @policy(
+						id: "%policyID%",
+						resource: "users"
+					) {
+						name: String
+						age: Int
+					}
+
+					type User3 @policy(
+						id: "%policyID%",
+						resource: "users"
+					) {
+						name: String
+						age: Int
+					}
+				`,
+
+				PolicyIDs: immutable.Some([]int{0, 0, 1}),
```

Comment:
suggestion: Having an index of the policyIDs is an nice addition. However, I find that this pattern is not very intuitive. Maybe if we made this a bit more generic it could even be useful for other dynamic assignments. My suggestion would be to change `PolicyIDs` in `SchemaUpdate` to a `Replace` field that holds a `map[string]any`. The outcome would give us something like this:
```go
testUtils.SchemaUpdate{
	Schema: `
		type Users1 @policy(
			id: "policy_0",
			resource: "users"
		) {
			name: String
			age: Int
		}

		type User2 @policy(
			id: "policy_0",
			resource: "users"
		) {
			name: String
			age: Int
		}

		type User3 @policy(
			id: "policy_1",
			resource: "users"
		) {
			name: String
			age: Int
		}
	`,

	Replace: immutable.Some(map[string]any{
        "policy_0": NewPolicyIndex(0),
        "policy_1": NewPolicyIndex(1),
    }),
}
```

Reply:
> ```go
> NewPolicyIndex(0)
> ```

While I do like the flexibility of the high-level suggestion, might more interesting to see if the usage with the util function `NewPolicyIndex(0)` is nicer (i am unsure atm).

Brainstorming a bit here, so the `Replace` command will do all the replacing and handle "all substitution" types (like  fetching policyIDs at Index x y z, or anything else in the future), for which we would need to define all these new index types? like one for policy and one for each other substitution types, etc?

But I guess I see the benefits are more than the cons above.

 If others are happy I can try to move to this proposed approach

@sourcenetwork/database-team Would be nice to hear prefs others have

Reply:
> for which we would need to define all these new index types

It's just a switch statement in the replace logic. We already have the `NewFooIndex` functions for docIDs and soon CIDs. One of the switch items could even be `string` to be able to substitute a manually defined string.

Reply:
I'd like to propose using Go templates as an alternative approach. It would make our schema definitions more readable and maintainable. 
Many developers are already familiar with native Go template syntax. We can easily extend it with custom template functions for our specific needs.
Here's a simple example:
```go
testUtils.SchemaUpdate{
    Schema: `
        type User1 @policy(
            id: "{{.Policy0}}",
            resource: "users"
        ) {
            name: String
            age: Int
        }

        type User2 @policy(
            id: "{{.Policy0}}",
            resource: "users"
        ) {
            name: String
            age: Int
        }

        type User3 @policy(
            id: "{{.Policy1}}",
            resource: "users"
        ) {
            name: String
            age: Int
        }
    `,
    TemplateData: map[string]string{
        "Policy0": "policy_0",
        "Policy1": "policy_1",
    },
}
```

We can even do some perversions with it (I'm not advocating):
```go
testUtils.SchemaUpdate{
    Schema: `
        {{- define "userType" -}}
        type {{.Name}} @policy(
            id: "{{.PolicyID}}",
            resource: "users"
        ) {
            name: String
            age: Int
        }
        {{- end -}}

        {{template "userType" dict "Name" "User1" "PolicyID" (policyIndex 0)}}
        {{template "userType" dict "Name" "User2" "PolicyID" (policyIndex 0)}}
        {{template "userType" dict "Name" "User3" "PolicyID" (policyIndex 1)}}
    `,
    TemplateFuncs: template.FuncMap{
        "policyIndex": func(i int) string {
            return fmt.Sprintf("policy_%d", i)
        },
        "dict": func(values ...any) (map[string]any, error) {
            if len(values)%2 != 0 {
                return nil, errors.New("dict requires pairs")
            }
            dict := make(map[string]any, len(values)/2)
            for i := 0; i < len(values); i += 2 {
                key, ok := values[i].(string)
                if !ok {
                    return nil, errors.New("dict keys must be strings")
                }
                dict[key] = values[i+1]
            }
            return dict, nil
        },
    },
}
```
It doesn't look very straightforward, at least not to the level we want to stick to in out tests. 
It will just give us a lot more room for creativity in the future.

Reply:
well, it can be `NewPolicyIndex(0)` or whatever else we decide.
I will update my proposal to make it complete.